### PR TITLE
Provides a default configuration for localhost out-of-the-box

### LINKED ISSUE #57: [Enhancement] | Enable the project to deploy standalone out-of-the-box
[Enhancement] | Enable the project to deploy standalone out-of-the-box
----------|----------
Epic Title | Developer-friendly codebase
T-shirt size: S(mall), M(edium), L(arge) | S
Story Points - Fibonacci (1,2,3,5,8,13) | 2
User Story | "As a developer, I want to easily deploy the application and quickly diagnose errors in deployment"
Details | The current sources provide a config.sample.py that is to be copied and renamed to config.py before the application can deploy to localhost. There are a number of issues with the current setup. When config.py is missing: * The app builds and deploys without errors  * Errors in the server are hidden within its container * Errors in the UI come out as CORS violations * An unmanaged default config is easily lost in forking
Acceptance Criteria/Tests | The code can be checked out and the application started locally without side effects or errors

### PR DESCRIPTION
**Pull request by kthblmfld**
The current sources provide a config.sample.py that is to be
copied and renamed to config.py before the application can deploy
to localhost. There are a number of issues with the current setup.
When config.py is missing:

  *  The app builds and deploys without errors
  *  Errors in the server are hidden within its container
  *  Errors in the UI come out as CORS violations
  *  An unmanaged default config is easily lost in forking

This change designates config.py as config-localhost.py and
references it by default in main.py in a constant,
CONFIG_OVERRIDE_FILE. This allows the application to deploy
out-of-the-box.

Resolves #57 
See also: #40

Signed-off-by: keithbloomfield <kthblmfld@gmail.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: python
+python:
```
Comment (kthblmfld) [inline .travis.yml]: Not sure why or how this change and the build health reporting in README are shown as missing in master (?)

File: README.md [hunk 2]
```diff
@@ -17,13 +20,8 @@ The easiest way to run these components is with
 [Docker](https://www.docker.com/what-docker). To start these components,
 first install Docker for your platform and clone this repo.
 
-Next, create a `server/config.py` file based off the example at
-`server/config.py.example`. Pay special attention to the secret keys at the
-bottom of the example file. While other settings have defaults and can be
-omitted, these keys have no defaults and _must_ be provided to run the server.
```
Comment (chrisspanton) [inline README.md]: This removes the (previously inaccurate) statement that sample keys are not provided and must be input. I'm concerned however that people may deploy without understanding that there is an insecure key used if there is no mention of it. Thoughts?

Comment (kthblmfld) [reply README.md]: I moved it to a section further down titled: "Deploying to Any Non-Localhost Server" and tried to elaborate further on the reasoning for having to supply your own config.yaml (why it should not be in VCS, etc)

Comment (chrisspanton) [reply README.md]: Evidence for why I should complete a review before commenting. ;) That said, the statement is still inaccurate, no? I believe there are defaults and they need not be provided to run the server.

**Comment by agoldingay** [comment]
All, need a second approver "pronto" before we can officially move this as "done". @mtn217 did the 1st approval. Need a 2nd approval, @chrisspanton ?

**Pull request merged**